packages/core/src/__stampers__/api/web/stamper.ts

```diff
@@ -0,0 +1,238 @@
+
+  private async getPrivateKey(publicKeyHex: string): Promise<CryptoKey | null> {
+    const db = await this.openDb();
+    return new Promise((resolve, reject) => {
+      const tx = db.transaction(DB_STORE, "readonly");
+      const store = tx.objectStore(DB_STORE);
+      const request = store.get(publicKeyHex);
+      request.onsuccess = () => {
+        db.close();
+        resolve(request.result || null);
+      };
+      request.onerror = () => {
+        db.close();
+        reject(request.error);
+      };
+    });
+  }
+
+  async listKeyPairs(): Promise<string[]> {
+    const db = await this.openDb();
+    return new Promise((resolve, reject) => {
+      const tx = db.transaction(DB_STORE, "readonly");
+      const store = tx.objectStore(DB_STORE);
+      const request = store.getAllKeys();
+      request.onsuccess = () => {
+        db.close();
+        resolve(request.result as string[]);
+      };
+      request.onerror = () => {
+        db.close();
+        reject(request.error);
+      };
+    });
+  }
+
+  async createKeyPair(externalKeyPair?: CryptoKeyPair): Promise<string> {
+    let privateKey: CryptoKey;
+    let publicKey: CryptoKey;
+    if (externalKeyPair) {
+      const extractable = (externalKeyPair.privateKey as any).extractable;
+      if (extractable !== false) {
+        throw new Error("Provided privateKey must be non-extractable.");
+      }
+      privateKey = externalKeyPair.privateKey;
+      publicKey = externalKeyPair.publicKey;
+    } else {
+      const keyPair = await crypto.subtle.generateKey(
+        { name: "ECDSA", namedCurve: "P-256" },
+        false, // Non-extractable private key
+        ["sign", "verify"],
+      );
+      privateKey = keyPair.privateKey;
+      publicKey = keyPair.publicKey;
+    }
+    const rawPubKey = new Uint8Array(
+      await crypto.subtle.exportKey("raw", publicKey),
+    );
+    const compressedPubKey = pointEncode(rawPubKey);
+    const compressedHex = uint8ArrayToHexString(compressedPubKey);
+    await this.storeKeyPair(compressedHex, privateKey);
+    return compressedHex;
+  }
+
+  async deleteKeyPair(publicKeyHex: string): Promise<void> {
+    const db = await this.openDb();
+    return new Promise((resolve, reject) => {
+      const tx = db.transaction(DB_STORE, "readwrite");
+      const store = tx.objectStore(DB_STORE);
+      store.delete(publicKeyHex);
+      tx.oncomplete = () => {
+        db.close();
+        resolve();
+      };
+      tx.onerror = () => reject(tx.error);
+    });
+  }
+
+  async clearKeyPairs(): Promise<void> {
+    const db = await this.openDb();
+    return new Promise((resolve, reject) => {
+      const tx = db.transaction(DB_STORE, "readwrite");
+      const store = tx.objectStore(DB_STORE);
+      store.clear();
+      tx.oncomplete = () => {
+        db.close();
+        resolve();
+      };
+      tx.onerror = () => reject(tx.error);
+    });
+  }
+
+  private async sign(payload: string, publicKeyHex: string): Promise<string> {
```

Comment:
nit: bare signature interfaces should always take bytes (Uint8Array) instead of string. Otherwise you end up having to make assumptions about string encoding, that's no bueno :D 

(it's not a big deal here because it's a private function, and there's only one caller)